mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -668,6 +668,48 @@ def SYCLCallOp : SYCL_Op<"call", [CallOpInterface]> {
   }];
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// accessor.get_pointer OPERATION
+////////////////////////////////////////////////////////////////////////////////
+
+def SYCLAccessorGetPointerOp
+    : SYCLMethodOpInterfaceImpl<"accessor.get_pointer", "AccessorType",
+                                ["get_pointer"]> {
+  let summary = "Call to accessor::get_pointer";
+  let description = [{
+    This operation represents a call to the accessor::get_pointer function.
+  }];
+
+  let arguments = (ins Arg<AccessorMemRef, "The accessor", [MemRead]>:$Acc,
+                       TypeArrayAttr:$ArgumentTypes,
+                       FlatSymbolRefAttr:$FunctionName,
```

Comment:
On a more general note: Our overall goals is to make the SYCL dialect less implementation-specific. 

Adding the (mangled) function name to the two new operations means they are implementation-specific again. 

I was wondering if this PR was a good opportunity to get started on our journey to a less implementation-specific dialect by leaving these attributes off and moving that functionality to the lowering of the operations.

Reply:
The `MangledFunctionName` is optional, and it can be generated without. As it is inherited from `SYCLMethodOpInterfaceImpl`, which contains function like 
```
    /// Return the callee of the generic SYCL call operation, this is required
    /// by the call interface.
    CallInterfaceCallable getCallableForCallee() {
      return (*this)->getAttrOfType<FlatSymbolRefAttr>(getMangledFunctionNameAttrName());
    }
```
I think is best to keep it similar to other `SYCLMethodOps` and remove them all together in a different PR.

Reply:
Ok, then it probably makes sense to have separate PR for that.

Reply:
I see Lukas' point, but it's also true that implementing `SYCLMethodOpInterfaceImpl` makes codegen for these operations free, so maybe we can keep this as is.